### PR TITLE
React 18 support

### PR DESCRIPTION
**Pull request by michaeltaranto**
### Changes
- Add React 18 to peer deps version list
- Bump as many react libraries to fix peer dep warnings
- No changes necessary to system from consumer side, with exception for `Dropdown` needing to wrap children in a `Fragment`.

### Site fixes
- Migrate site from `react-head` to `react-helmet-async` due to TS incompatibility, but it also comes with its own types.

### Testing fixes
- Migrate some tests to use `@testing-library/user-event` for their interactions.
- The `@testing-libary/react-hooks` library is now built into the `@testing-library/react` package.
- Exclude props starting with `__@` from the contract snapshot, avoiding the ever incrementing indexes in the `filterSuggestions` snapshot.

### Additional note
There are still a handful of deps that complain about React peer dep version, but are functionally ok. Will update as new versions become available.

### CODE REVIEW

File: site/src/client.tsx [hunk 1]
```diff
@@ -38,16 +38,16 @@ const ScrollManager = ({ children }: { children: ReactNode }) => {
 export default (app: RenderContext) => {
   initUpdates(new Date(app.renderDate), app.versionMap, app.currentVersion);
 
-  hydrate(
-    <HeadProvider>
+  hydrateRoot(
+    document.getElementById('app')!,
```
Comment (benjervis) [inline site/src/client.tsx:42]: Is this `!` a real thing, or typo?

File: site/src/render.tsx [hunk 2]
```diff
@@ -76,23 +76,22 @@ const skuRender: Render<RenderContext> = {
...
     ).join('');
+    const { helmet } = helmetContext;
 
     return dedent`
       <!doctype html>
       <html lang="en">
         <head>
           ${__experimentalDarkMode__}
-          ${
-            // @ts-expect-error
```
Comment (benjervis) [inline site/src/render.tsx:92]: 👏

File: lib/components/Stepper/Stepper.test.tsx [hunk 3]
```diff
@@ -190,6 +190,8 @@ describe('Stepper', () => {
       await userEvent.keyboard('{arrowleft}');
       expect(secondStep).toHaveFocus();
       await userEvent.tab();
+      expect(thirdStep).toHaveFocus();
```
Comment (benjervis) [inline lib/components/Stepper/Stepper.test.tsx:193]: Did we change something to give the Stepper an extra step?

**Review by benjervis — APPROVED** [review]
🎉

File: site/src/client.tsx (hunk 1 above)
Comment (michaeltaranto) [reply site/src/client.tsx:42]: Haha, nah i got this from their [migration guide](https://reactjs.org/blog/2022/03/08/react-18-upgrade-guide.html#updates-to-client-rendering-apis). I originally added a null check on the `getElementById` call, but then spotted this in their docs.

File: lib/components/Stepper/Stepper.test.tsx (hunk 3 above)
Comment (michaeltaranto) [reply lib/components/Stepper/Stepper.test.tsx:193]: I havent been able to put my finger on this one, why this is passing in master has me confused as the test now replicates what happens in the browser (both production and this branch).

Will have a poke around and see if i can provide some more clarity.

File: site/src/client.tsx (hunk 1 above)
Comment (benjervis) [reply site/src/client.tsx:42]: Oh the null thing, forgot about types 😅

**Pull request merged**